x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/aggregations/bucket/histogram/HistoBackedHistogramAggregator.java

```diff
@@ -86,9 +86,10 @@ public void collect(int doc, long owningBucketOrd) throws IOException {
                             } else {
                                 collectBucket(sub, doc, bucketOrd);
                             }
-                            // We have added the document already. We should increment doc_count by count - 1
-                            // so that we have added it count times.
-                            incrementBucketDocCount(bucketOrd, count - 1);
+                            // We have added the document already and we have incremented bucket doc_count
+                            // by _doc_count times. To compensate for this, we should increment doc_count by
+                            // (count - _doc_count) so that we have added it count times.
+                            incrementBucketDocCount(bucketOrd, count - docCountProvider.getDocCount(doc));
```

Comment:
No matter what `_doc_count` is, it was previously added to the `bucket_count` via the `collectBucket` methods. It could be ANY number. 

Consequently, this `incrementBucketDocCount` may actually be decrementing the `bucket_count` to adjust for the difference. 

I think this is fine. 

The other potential solution is to override `collectBucket...` so the bucket count is not incremented. But that may prove too complicated.

I think this is good solution for now 👍 

Somebody else from the aggs team should take a look as well.